src/hasSubObject.ts

```diff
@@ -0,0 +1,68 @@
+import { isDeepEqual } from "./isDeepEqual";
+import { purry } from "./purry";
+import type { Merge } from "./type-fest/merge";
+
+/**
+ * Checks if `subObject` is a sub-object of `object`, which means for every
+ * property and value in `subObject`, there's the same property in `object`
+ * with an equal value. Equality is checked with `isDeepEqual`.
+ *
+ * @param data - The object to test.
+ * @param subObject - The sub-object to test against.
+ * @signature
+ *    R.hasSubObject(data, subObject)
+ * @example
+ *    R.hasSubObject({ a: 1, b: 2, c: 3 }, { a: 1, c: 3 }) //=> true
+ *    R.hasSubObject({ a: 1, b: 2, c: 3 }, { b: 4 }) //=> false
+ *    R.hasSubObject({ a: 1, b: 2, c: 3 }, {}) //=> true
+ * @dataFirst
+ * @category Object
+ */
+export function hasSubObject<T, S extends Partial<T>>(
+  data: T,
+  subObject: S,
+  // @ts-expect-error: typescript doesn't infer merges correctly
```

Comment:
Can you add the typescript error too? that way when a change in typescript causes the error to not be relevant, or when someone tries to fix the ts error they know what was suppressed here. (I wish typescript would support saying what error is expected and not this generic disabler, like `eslint-disable` supports)

```ts
// @ts-expect-error [tsXXXX]: ...
```